File: zoomx/src/main/java/com/zoomx/zoomx/model/HeaderViewModel.java
package com.zoomx.zoomx.model;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Ahmed Fathallah on 11/20/2017.
 */
public class HeaderViewModel {

    private Map<String, String> headersMap;

    public HeaderViewModel() {
        this.headersMap = new HashMap<>();
    }

    public HeaderViewModel(Map<String, String> headersMap) {
        this.headersMap = headersMap;
    }

    public void addHeader(String name, String value) {
        headersMap.put(name, value);
    }

    public Map<String, String> getHeadersMap() {
        return headersMap;
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/model/HeaderViewModel.kt
package com.zoomx.zoomx.model

import java.util.*

/**
 * Created by Ahmed Fathallah on 11/20/2017.
 */
class HeaderViewModel(var headersMap: MutableMap<String, String> = HashMap()) {

    fun addHeader(name: String, value: String) {
        headersMap[name] = value
    }

    override fun toString(): String {
        return headersMap.map { it.value }.reduce { sum, element -> sum + "\n" + element }
    }

}
